docs/specs/oidc.md

```diff
@@ -405,3 +406,82 @@ Silent Authentication
 1. SDK reads the result message, logout if the result indicates IdP Session is invalid
 1. If authorization code request result is successful, send the token request to Authgear with code + code verifier
 1. Authgear returns token response to SDK with id token + new access token
+
+## First-party Clients
+
+OAuth protocol allows user to delegate access to clients, so that clients can
+act on behalf of the user.
+
+Usually, OAuth clients are third-party clients, with limited trust:
+- User need to give consent to a client before clients is allowed access.
+- Privileged user operations (e.g. change password) is usually not exposed
+  through APIs accessible through OAuth protocol.
+
+However, developers may want to give first-party clients full trust:
```

Comment:
Similarly, for the clients vs applications wordings, shall we change these sentences (and other related similarly) to:

"However, developers may want to allow clients full trust as a first-party application:"

something like that.